assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidgetGA4.js

```diff
@@ -0,0 +1,362 @@
+/**
+ * DashboardOverallPageMetricsWidgetGA4 component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x, sprintf, _n } from '@wordpress/i18n';
+import { isURL } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS,
+} from '../../datastore/constants';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
+import { Grid, Row, Cell } from '../../../../material-components/layout';
+import PreviewBlock from '../../../../components/PreviewBlock';
+import DataBlock from '../../../../components/DataBlock';
+import Sparkline from '../../../../components/Sparkline';
+import SourceLink from '../../../../components/SourceLink';
+import whenActive from '../../../../util/when-active';
+import { generateDateRangeArgs } from '../../util/report-date-range-args';
+import { calculateChange, getURLPath, stringToDate } from '../../../../util';
+import parseDimensionStringToDate from '../../util/parseDimensionStringToDate';
+import WidgetHeaderTitle from '../../../../googlesitekit/widgets/components/WidgetHeaderTitle';
+import useViewOnly from '../../../../hooks/useViewOnly';
+const { useSelect, useInViewSelect } = Data;
+
+/**
+ * Fetches Analytics 4 report data and state for the Overall Page Metrics widget.
+ *
+ * @since n.e.x.t
+ *
+ * @typedef {Object} OverallPageMetricsReport
+ * @property {Object|undefined}   report     - Analytics report data if exists, otherwise undefined.
+ * @property {Object}             dates      - Object containing the report date ranges.
+ * @property {string}             serviceURL - Link to relevant Google Analytics page for the report.
+ * @property {boolean}            isLoading  - Loading status for report.
+ * @property {(Object|undefined)} error      - Error object if exists, otherwise undefined.
+ * @return {OverallPageMetricsReport} Analytics report data and state.
+ */
+function useOverallPageMetricsReport() {
```

Comment:
Can you remind me why we use such a unique approach for this and the original widget? This function is not a react component, but it uses hooks that are not intended for usage in non-component functions. Plus, we don't use this approach in other components that have many select hooks. I think we should get rid of this function because it seems to be an anti-pattern. Feel free to rework the original widget as well.

Reply:
I'm not sure why you describe this as an anti-pattern - this is simply a [custom hook](https://reactjs.org/docs/hooks-custom.html), and it's perfectly legitimate to call other hooks from a custom hook.

In terms of the why, this was one of my first PRs to the project. It still does seem to me a logical and elegant way to encapsulate the data fetching side of things, separately from the rendering logic of the component. As it happens, looking back in history, you actually reviewed the [PR](https://github.com/google/site-kit-wp/pull/4294) for this one :sweat_smile: 

That said, I do recognise that it's a bit outside of the usual patterns we apply.

It feels a bit out of scope to refactor the original component like this. Bearing in mind the above, do you still want me to rewrite this?

Reply:
Update: In the meantime, I've extracted this hook into its own function. Still happy enough to rewrite it though, if you prefer.

Reply:
> I'm not sure why you describe this as an anti-pattern - this is simply a [custom hook](https://reactjs.org/docs/hooks-custom.html), and it's perfectly legitimate to call other hooks from a custom hook.

Apologies, for some reason, I misunderstood this function and oversight that it is a custom hook 🤦. 

As to "anti-pattern", I meant that it is not consistent with the pattern that we use in all other components that have many different hooks, we don't create a custom hook for each of them.

> As it happens, looking back in history, you actually reviewed the [PR](https://github.com/google/site-kit-wp/pull/4294) for this one 😅

Looks like I looked at it more accurately that time 😄 

> It feels a bit out of scope to refactor the original component like this. 

Up to you. It's okay to me if we update only the new widget.

> Bearing in mind the above, do you still want me to rewrite this?

Yes, please. Let's make it work the same way as other components do.

Reply:
No problem - I'll update them both :+1:

Reply:
Ok, that's done.

Reply:
Thanks, @techanvil.